test/test_torch.py

```diff
@@ -16330,16 +16329,148 @@ def test_svd_tall_some_col_maj(self, device, dtype):
+    'rsqrt': ufunc_meta(is_floating=True, complex_on=('cpu')),
+    'trunc': ufunc_meta(is_floating=True),
+    'erf': ufunc_meta(is_floating=True, complex_on=()),
+    'erfc': ufunc_meta(is_floating=True, complex_on=()),
+    'erfinv': ufunc_meta(is_floating=True, complex_on=()),
+    'lgamma': ufunc_meta(is_floating=True, complex_on=()),
+    'digamma': ufunc_meta(is_floating=True, complex_on=()),
+    'sigmoid': ufunc_meta(is_floating=True, complex_on=('cpu'))
+}
+
+# Creates and adds a test of the given unary floating ufunc's type promotion
+# Unary floating ufuncs promote tensors of bool and integral types to
+# the default scalar type.
+def generate_unary_floating_ufunc_promo_test(cls, op_str):
+    int_types = [torch.bool, torch.uint8, torch.int8, torch.int16,
+                 torch.int, torch.long]
+    float_types = [torch.float, torch.double]
+    float_types_cuda = [torch.half, torch.float, torch.double]
+    complex_types = [torch.complex64, torch.complex128]
+    default_types = [torch.float, torch.double]
+    default_types_cuda = [torch.half, torch.float, torch.double]
+
+    def test_fn(self, device):
+        op = getattr(torch, op_str)
+
+        my_default_types = default_types
+        my_float_types = float_types
+        if self.device_type == 'cuda':
+            my_default_types = default_types_cuda
+            my_float_types = float_types_cuda
+        if self.device_type in unary_ufuncs[op_str].complex_on:
+            my_float_types.extend(complex_types)
+
+        prev_def_type = torch.get_default_dtype()
+        try:
+            for in_type, def_type in product(int_types, my_default_types):
+                torch.set_default_dtype(def_type)
+                t = torch.tensor((1,), device=device, dtype=in_type)
+                self.assertEqual(op(t).dtype, def_type)
+        finally:
+            torch.set_default_dtype(prev_def_type)
+
+        for in_type in my_float_types:
+            t = torch.tensor((1,), device=device, dtype=in_type)
+            self.assertEqual(op(t).dtype, in_type)
+
+        # These ops only support out= calls on CPU devices
+        only_out_on_cpu = ['atan', 'cos', 'cosh', 'erf', 'erfc', 'exp', 'tan', 'tanh']
+
+        # Test for out= code paths
+        for in_type in my_float_types + int_types:
+            # Floating type promotion for out= calls on ops listed in only_out_on_cpu
+            # not supported on cuda devices
+            if (op_str in only_out_on_cpu and self.device_type == 'cuda'):
+                continue
+            t = torch.tensor((1,), device=device, dtype=in_type)
+            for out_type in my_float_types:
+                # Floating type promotion does not support out= calls for input as bool tensors
+                # and output as complex tensors
+                if (in_type == torch.bool and out_type in complex_types):
+                    continue
+                out_t = torch.tensor((), device=device, dtype=out_type)
```

Comment:
Should this have a size of 1? torch.empty((1,), ...)?

Reply:
Done! https://github.com/pytorch/pytorch/pull/33322/commits/49ec5a40809f4f496b2850ab6a45579f92c8f520